scripts/Muon/GUI/Common/results_tab_widget/results_tab_model.py

```diff
@@ -148,19 +152,20 @@ def create_results_table(self, log_selection, results_selection):
                         log_value = np.nan
                     row_dict.update({log_name: log_value})
             # fit parameters
-            parameter_dict = fit.parameter_workspace.workspace.toDict()
-            for name, value, error in zip(parameter_dict['Name'],
-                                          parameter_dict['Value'],
-                                          parameter_dict['Error']):
-                row_dict.update({name: value})
-                if name not in RESULTS_TABLE_COLUMNS_NO_ERRS:
-                    row_dict.update({name + 'Error': error})
+            for param_name in fit_parameters.names():
+                row_dict.update({param_name: fit_parameters.value(param_name)})
+                if _param_error_should_be_displayed(param_name):
+                    row_dict.update({
+                        _error_column_name(param_name):
+                        fit_parameters.error(param_name)
+                    })
 
             try:
                 results_table.addRow(row_dict)
             except ValueError:
                 msg = "Incompatible fits for results table:\n"
-                msg += " fit 1 and {} have a different parameters".format(position + 1)
+                msg += " fit 1 and {} have a different parameters".format(
```

Comment:
could we return the names of the workspaces? If this is a lot of work, we will do it later.

Reply:
Good point. I did this in a rush last week. That should be easy.